upup/pkg/fi/cloudup/bootstrapchannelbuilder/bootstrapchannelbuilder.go

```diff
@@ -531,6 +532,23 @@ func (b *BootstrapChannelBuilder) buildAddons(c *fi.ModelBuilderContext) (*chann
 				})
 			}
 		}
+
+		if b.UseServiceAccountIAM() {
+			awsModelContext := &awsmodel.AWSModelContext{
+				KopsModelContext: b.KopsModelContext,
+			}
+
+			serviceAccountRoles := []iam.Subject{&clusterautoscaler.ServiceAccount{}}
```

Comment:
We should refactor this to avoid copy-pasta. Pull `serviceAccountRoles` to the beginning of `buildAddons()`, push `awsModelContext` and the for loop to the end of `buildAddons()`, and have this block of code and the ones for dns-controller and aws-load-balancer-controller just append to `serviceAccountRoles`